src/plugin_discovery/find_plugin_specs.js

```diff
@@ -46,6 +47,7 @@ export function findPluginSpecs(settings, config = defaultConfig(settings)) {
     ...config.get('plugins.paths').map(createPackAtPath$),
     ...config.get('plugins.scanDirs').map(createPacksInDirectory$)
   )
+    .distinct(({ pack, error }) => error || (pack && realpathSync(pack.getPath())))
```

Comment:
**Nitpick**: It's not super-easy to read the intention here, maybe:

```js
.distinct(pack => dedupeOnAbsolutePath(pack))

// ...

// Returns a value that can be checked for uniqueness using `===`
// to determine if we want to dedupe or include duplicates.
function dedupeOnAbsolutePath({ pack, error }) {
  if (error !== undefined) {
    // For errors we pass the reference, as all errors are unique and should not
    // be deduped.
    return error;
  }

  // Otherwise we return the absolute path, as we don't want to include projects with
  // the same path multiple times.
  return realpathSync(pack.getPath())
}
```

(can both `error` and `pack` be `undefined` at the same time?)

It's way more wordy, but hopefully easier to grasp later. Not sure though, as the naming isn't particularly good. I'm fine with keeping what's there already too.

Reply:
👍